File: matlab/gaussSamp.m
function y = gaussSamp(Sigma, numSamps)

% GAUSSSAMP Sample from a Gaussian with a given covariance.

% NDLUTIL

[U, V] = eig(Sigma);
dims = size(Sigma, 1);
y = randn(numSamps, dims);
y = y*diag(sqrt(diag(V)));
y = y*U';

File: R/gaussSamp.R
gaussSamp <- function(Sigma, numSamps) {

  eigVecs = eigen(Sigma)
  U = eigVecs$vectors; V = eigVecs$values
  dims = dim(Sigma)[1]
  y = matrix(rnorm(numSamps*dims), numSamps, dims) #randn(numSamps, dims)
  V[V<0] = as.complex(V[V<0])
  y = y %*% diag(sqrt(V)) %*% t(U)

  return (y)
}